### PR TITLE
ROX-11513, ROX-14077: Do not run QA part 2 if part 1 fails.

### PR DESCRIPTION
**Pull request by porridge**
## Description

See the ticket for the rationale.
TL;DR: a failure in part 1 sometimes leaves environment in a state unpalatable to part 2

## Checklist
- [x] Investigated and inspected CI test results
- [x] ~Unit test and regression tests added~
- [x] ~Evaluated and added CHANGELOG entry if required~
- [x] ~Determined and documented upgrade steps~
- [x] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

A [followup PR](https://github.com/stackrox/stackrox/pull/4254) with a change that breaks part1 and verify that part2 is skipped.

### CODE REVIEW

**Review by gavin-stackrox — APPROVED** [review]
Yeah, I think this makes sense (assuming it works as advertised). Similar to the rationale for enabling --fail-fast for nightlies.
